django/template/defaulttags.py

```diff
@@ -1167,6 +1168,47 @@ def now(parser, token):
     return NowNode(format_string, asvar)
 
 
+@register.simple_tag(takes_context=True)
+def query_string(context, query_dict=None, **kwargs):
+    """
+    Add, remove, and change parameters of a `QueryDict` and return the result
+    as a querystring. If the `query_dict` argument is not provided,
+    `request.GET` will be used.
+
+    For example::
+
+        {% query_string foo=3 %}
+
+    To remove a key::
+
+        {% query_string foo=None %}
+
+    To use with pagination::
+
+        {% query_string page=page_obj.next_page_number %}
+
+    A custom `QueryDict` can also be used:
+
+        {% query_string my_query_dict foo=3 %}
+    """
+    if query_dict:
+        query_dict = query_dict.copy()
+    else:
+        query_dict = context.request.GET.copy()
```

Comment:
It probably makes sense to check `is None` to avoid a case where an intentionally empty `query_dict` is passed falling back to `request.GET`. We should have a test for this case.

```suggestion
    if query_dict is None:
        query_dict = context.request.GET
    query_dict = query_dict.copy()
```

We should probably also have a another test where we actually pass in a custom, non-empty `QueryDict` instance as we only seem to have `query_string08` which explicitly passes in `request.GET`.

Reply:
I added this test already when addressing the other test.

This change seems reasonable to me, pushed this as well 👍🏻